parser/parser_rules.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015 The Serulian Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package parser
+
+// consumeTopLevel attempts to consume the top-level constructs of a Serulian source file.
+func (p *sourceParser) consumeTopLevel() AstNode {
+	rootNode := p.startNode(NodeTypeFile)
+	defer p.finishNode()
+
+	// Start at the first token.
+	p.consumeToken()
+
+Loop:
+	for {
+		switch {
+
+		// imports.
+		case p.isKeyword("import") || p.isKeyword("from"):
+			p.currentNode().Connect(NodePredicateChild, p.consumeImport())
+
+		// Whitespace.
+		case p.isToken(tokenTypeNewline):
+			// Skip the whitespace.
+			continue
+
+		// EOF.
+		case p.isToken(tokenTypeEOF):
+			// If we hit the end of the file, then we're done but not because of an expected
+			// rule.
+			p.emitError("Unexpected EOF at root level: %v", p.currentToken.position)
+			break Loop
+
+		case p.isToken(tokenTypeError):
+			break Loop
+
+		default:
+			p.emitError("Unknown token at root level: %v", p.currentToken)
+			break Loop
+
+		}
+
+		if p.isToken(tokenTypeEOF) {
```

Comment:
The imports case will reach it, as will other cases in the future where we expect either a newline OR an EOF